src/gui/qgswidgetstatehelper_p.h

```diff
@@ -0,0 +1,38 @@
+/***************************************************************************
+  qgswidgetstatehelper_p.h - QgsWidgetStateHelper
+
+ ---------------------
+ begin                : 3.12.2017
+ copyright            : (C) 2017 by nathan
+ email                : [your-email-here]
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSWIDGETSTATEHELPER_P_H
+#define QGSWIDGETSTATEHELPER_P_H
+
+#include <QMap>
+#include <QObject>
+
+#define SIP_NO_FILE
+
+class QgsWidgetStateHelper : public QObject
```

Comment:
Docs please!

Reply:
You and your bloody docs ;) Nah all good already done in a incoming commit.